### PR TITLE
Bug 2034957: update openshift-apiserver to kube 1.23.1

### PR DESCRIPTION
**Pull request by sanchezl**
- use go 1.17
- use kube 1.23.1
- update generated code
- fix logging flags
- use ConfigForResource

### CODE REVIEW

File: pkg/image/apiserver/registry/image/etcd/etcd_test.go [hunk 1]
```diff
@@ -25,7 +26,8 @@ import (
 func newStorage(t *testing.T) (*REST, *etcdtesting.EtcdTestServer) {
 	server, etcdStorage := etcdtesting.NewUnsecuredEtcd3TestClientServer(t)
 	etcdStorage.Codec = legacyscheme.Codecs.LegacyCodec(schema.GroupVersion{Group: "image.openshift.io", Version: "v1"})
-	imageRESTOptions := generic.RESTOptions{StorageConfig: etcdStorage, Decorator: generic.UndecoratedStorage, DeleteCollectionWorkers: 1, ResourcePrefix: "images"}
+	etcdStorageConfigForImages := &storagebackend.ConfigForResource{Config: *etcdStorage, GroupResource: schema.GroupResource{Group: "image.openshift.io", Resource: "images"}}
```
Comment (adambkaplan) [inline pkg/image/apiserver/registry/image/etcd/etcd_test.go:29]: /cc @dmage

File: pkg/route/apiserver/registry/route/etcd/etcd_test.go [hunk 2]
```diff
@@ -57,7 +58,8 @@ func (t *testSAR) Create(_ context.Context, subjectAccessReview *authorizationap
 func newStorage(t *testing.T, allocator routeinterfaces.RouteAllocator) (*REST, *etcdtesting.EtcdTestServer) {
 	server, etcdStorage := etcdtesting.NewUnsecuredEtcd3TestClientServer(t)
 	etcdStorage.Codec = legacyscheme.Codecs.LegacyCodec(schema.GroupVersion{Group: "route.openshift.io", Version: "v1"})
-	restOptions := generic.RESTOptions{StorageConfig: etcdStorage, Decorator: generic.UndecoratedStorage, DeleteCollectionWorkers: 1, ResourcePrefix: "routes"}
+	etcdStorageConfigForRoutes := &storagebackend.ConfigForResource{Config: *etcdStorage, GroupResource: schema.GroupResource{Group: "route.openshift.io", Resource: "routes"}}
```
Comment (adambkaplan) [inline pkg/route/apiserver/registry/route/etcd/etcd_test.go:61]: /cc @Miciah

**Review by adambkaplan — COMMENTED** [review]
Mostly looks good - I have a few questions regarding the `main.go` changes. I have also mentioned the team leads impacted by the etcd config change.

Personally, I'm okay if the go 1.17 changes and k8s vendor changes are squashed into a single commit, since the `go.mod` change by itself is a broken commit. The k8s vendor commit includes the new require section in `go.mod` to resolve transitive dependencies.

File: cmd/openshift-apiserver/main.go [hunk 3]
```diff
@@ -1,36 +1,29 @@
...
 
 import (
-	goflag "flag"
 	"fmt"
 	"math/rand"
 	"os"
 	"runtime"
 	"time"
 
+	"github.com/openshift/library-go/pkg/serviceability"
```
[the remark this answers is not in the document]
Comment (sanchezl) [reply cmd/openshift-apiserver/main.go:10]: Fixed.

File: cmd/openshift-apiserver/main.go [hunk 4]
```diff
@@ -39,6 +32,10 @@ func main() {
 	}
 
 	command := NewOpenshiftApiServerCommand(stopCh)
+	// add logging-related flags
+	command.SetGlobalNormalizationFunc(utilflag.WarnWordSepNormalizeFunc)
+	logs.AddFlags(command.PersistentFlags())
+
 	if err := command.Execute(); err != nil {
 		fmt.Fprintf(os.Stderr, "%v\n", err)
 		os.Exit(1)
```
[the remark this answers is not in the document]
Comment (sanchezl) [reply cmd/openshift-apiserver/main.go:41]: I was getting some `go.mod` errors and thought it wasn't yet available in `1.23.1`. It was a problem in my environment. Fixed.

**Comment by p0lyn0mial** [comment]
I'm tagging this PR just in case you decide to merge.

**Pull request merged**